src/config.rs

```diff
@@ -364,6 +381,17 @@ pub fn kernel_zram_option(root: &Path) -> Option<bool> {
     }
 }
 
+pub fn eval_size_expression(expr: &str, ram_size: u64) -> Result<f64> {
+    let x = Expr::new(expr)
+        .value("ram", ram_size)
+        .exec()
+        .with_context(|| format!("Expression \"{}\" cannot be evaluated", expr))?;
+    match x.as_f64() {
+        Some(x) => Ok(x),
+        None => Err(anyhow!("Expression evaluated to \"{}\", which is not a number", x))
```

Comment:
`x.as_f64().or_else(|x| anyhow!("", x))`

Reply:
That doesn't seem to work. What about this:
```rust
x.as_f64().
        with_context(|| format!("{}: expression evaluated to \"{}\", which is not a number", key, x))
```